### ISSUE TITLE
Use `tesh` for integration testing

### ISSUE THREAD

**Issue by jamesaoverton** [ontodev/nanobot.rs#14]
I would like to try using <https://github.com/OceanSprint/tesh> for testing our documentation. It requires Python, and can be installed with `pip install tesh`.

I created a [`test-init`](https://github.com/ontodev/nanobot.rs/tree/test-init) branch with a `doc/init.md` file that uses `tesh`. Code blocks that should be run through `tesh` are marked with `tesh-session="foo"`. Block with the same session name are run consecutively. Tests are executed in a new shell inside a temp directory. The new shell doesn't know about the repository directory, but it does inherit the `PATH`. In order for `tesh` to know about the `nanobot` binary, I did this:

```console
$ export PATH="${PATH}:$(pwd)/target/release/"
$ tesh doc/
results
```

By default, if a test fails, `tesh` drops you into a `bash` session in the temporary directory. For automated testing, use `tesh doc/ --debug false`.

**Comment by jamesaoverton** [comment]
`tesh` requires Python and then `pip install tesh` and the PATH updated. In the `Makefile` #12 this worked for me:

```make
target/debug/nanobot:
    cargo build

.PHONY: tesh
tesh: target/debug/nanobot
    python3 -m venv .venv; . .venv/bin/activate; pip install tesh; PATH="$${PATH}:$$(pwd)/target/debug";  tesh --debug false doc/
```

**Comment by jamesaoverton** [comment]
Closed by #12
